# IMPORTS

In [1]:
import pandas as pd
import os

In [2]:
pd.set_option('display.max_colwidth', -1) # Setting this so we can see the full content of cells

# Read and analyse the Dataset which contains the tweets we collected using Tweepy API

In [3]:
df = pd.read_csv("Tweetlabelled.csv")

In [4]:
df.head(5)

ID           Date- Time              User_Name  \
0  1289651266293919744  2020-08-01 19:56:37  @KokabMak               
1  1289651264972902401  2020-08-01 19:56:36  Sunny Phat              
2  1289651239496515586  2020-08-01 19:56:30  Area67                  
3  1289651238988988416  2020-08-01 19:56:30  Tony                    
4  1289651191597551616  2020-08-01 19:56:19  Coby Jay 3Fingers XIX   

                                                                                                                                                                                                                                                                                                     Tweet  \
0  @AdanechAbiebie #DejeneTefa were healthy when your military arrest him from his home , then intentionally infected by Corona virus. \nMorethan 200 persons are held in a single class across oromia.. \nPuppet AG                                                                                         
1  @RepMikeRogersAL She's not playing. The House sent over a bill in May, addressing all these issues. Grim Reaper Mitch buried it. Now he can't even figure out why 15-20 of his GOP Sens. refuse to vote for any bill. 160,000 Americans will soon be dead from corona virus, Pence and Trump: maskless.   
2  It's  been a tough time out there for a lot of people due to Corona virus, we do have a fantastic opportunity available as a branch manager at our Hailsham branch. If you or anyone you know would be interested in joining our fantastic team please contact us                                         
3  The Corona virus test feels weirrrrd bro😳                                                                                                                                                                                                                                                                 
4  @ColossalTree @amuse @NPR @realDonaldTrump These malaria pills. In 2005 Fauci suggested using oxychloroquine for a strain of the corona virus. F**** off . By the way, he never advised injecting disinfectant. You really aren't too bright.                                                             

  Sentiment  
0  positive  
1  negative  
2  positive  
3  neutral   
4  positive

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30528 entries, 0 to 30527
Data columns (total 5 columns):
ID            30528 non-null int64
Date- Time    30528 non-null object
User_Name     30525 non-null object
Tweet         30528 non-null object
Sentiment     30528 non-null object
dtypes: int64(1), object(4)
memory usage: 1.2+ MB


# Check if any duplicate tweets are present in the dataset

        based on tweet id check if any duplicate tweets exist. 
        if two tweets share the same tweet id , they are considered as duplicates and onne of those are displayed.

In [7]:
duplicateRowsDF = df[df.duplicated(['ID'])]

In [8]:
duplicateRowsDF

ID           Date- Time  \
289    1289651244668080128  2020-08-01 19:56:31   
290    1289651177051717633  2020-08-01 19:56:15   
905    1289635774023909376  2020-08-01 18:55:03   
1202   1289650985472872449  2020-08-01 19:55:30   
1532   1289651230072025088  2020-08-01 19:56:28   
1704   1289651472901136384  2020-08-01 19:57:26   
1982   1289652811945390082  2020-08-01 20:02:45   
2025   1289651491914092550  2020-08-01 19:57:30   
2157   1289634290897006592  2020-08-01 18:49:09   
2163   1289633499637374976  2020-08-01 18:46:01   
2396   1289565774420389889  2020-08-01 14:16:54   
2692   1289499318865367041  2020-08-01 09:52:50   
2702   1289465440729612289  2020-08-01 07:38:12   
2815   1289649127588950017  2020-08-01 19:48:07   
2837   1289646391657029632  2020-08-01 19:37:14   
3393   1289648800219435009  2020-08-01 19:46:49   
3411   1289640021008367617  2020-08-01 19:11:56   
3464   1289655189968949248  2020-08-01 20:12:12   
3466   1289652901560819712  2020-08-01 20:03:06   
3499   1289623434092109825  2020-08-01 18:06:01   
3507   1289619374991486976  2020-08-01 17:49:53   
3513   1289615758117371905  2020-08-01 17:35:31   
3581   1289647391205777408  2020-08-01 19:41:13   
3832   1289655790547030017  2020-08-01 20:14:35   
3861   1289652375662333958  2020-08-01 20:01:01   
3862   1289652136339369984  2020-08-01 20:00:04   
4133   1289653238824710144  2020-08-01 20:04:27   
4136   1289652375662333958  2020-08-01 20:01:01   
4175   1289644360309514240  2020-08-01 19:29:10   
4317   1289658159972270080  2020-08-01 20:24:00   
...                    ...                  ...   
29534  1324044292827041792  2020-11-04 17:42:13   
29538  1324044288657969158  2020-11-04 17:42:12   
29549  1324044267627700229  2020-11-04 17:42:07   
29552  1324044265085960193  2020-11-04 17:42:07   
29556  1324044257699729408  2020-11-04 17:42:05   
29557  1324044256907055105  2020-11-04 17:42:05   
29569  1324044241903980544  2020-11-04 17:42:01   
29590  1324044185368961024  2020-11-04 17:41:48   
29591  1324044184139952128  2020-11-04 17:41:47   
29593  1324044176707694592  2020-11-04 17:41:45   
29603  1324044161448894465  2020-11-04 17:41:42   
29618  1324044141106323457  2020-11-04 17:41:37   
29730  1324044165030711296  2020-11-04 17:41:43   
29731  1324044155925024770  2020-11-04 17:41:40   
29844  1324044292827041792  2020-11-04 17:42:13   
29845  1324044292751577091  2020-11-04 17:42:13   
29854  1324044285034070018  2020-11-04 17:42:11   
29866  1324044257699729408  2020-11-04 17:42:05   
29870  1324044251538350082  2020-11-04 17:42:03   
29876  1324044241903980544  2020-11-04 17:42:01   
29879  1324044236665221121  2020-11-04 17:42:00   
29891  1324044218147524608  2020-11-04 17:41:55   
29905  1324044185318662145  2020-11-04 17:41:47   
29910  1324044176707694592  2020-11-04 17:41:45   
29911  1324044175420149763  2020-11-04 17:41:45   
29915  1324044171494203392  2020-11-04 17:41:44   
29921  1324044161448894465  2020-11-04 17:41:42   
30030  1324044287730995200  2020-11-04 17:42:12   
30035  1324044062056452096  2020-11-04 17:41:18   
30098  1324042471668473856  2020-11-04 17:34:59   

                           User_Name  \
289    bc                              
290    bc                              
905    thomas Jr. 🛠                    
1202   Steve Harrop                    
1532   George Macpherson               
1704   Dave Bryant                     
1982   jkonowe                         
2025   Martino Dembini                 
2157   Siddharth Gupta                 
2163   Arivanantham Rangasa            
2396   Tania Bhattacharjee             
2692   RAHUL #Asim Riaz Fan            
2702   Lopa                            
2815   Febre Mania                     
2837   jesus was a juggalo             
3393   Filipe Orlando                  
3411   Your logo guy ❁                 
3464   SuperSport 🏆                    
3466   DamnitDan                       
3499   Cristiano Ronaldo Motivation™   
3507   Whirling_Dervi

        if we scroll till the end of the output , we see we have around 1078 duplicate entries

# Delete duplicate tweets keep only one amomg the duplicates.

    Sort the tweets based on id , so tweets with same id will be placed one after the other.
    Next using drop_duplicates, we drop/ delete rows which have same tweet id.
    Out of the duplicate tweets only the first tweet is ket rest are dropped 
    lastly 'inplace = True' means the chances made will be reflected in the dataframe or the duplicate rows are deleted
    permanently from the dataframe 'df'.

In [9]:
df.sort_values("ID", inplace = True) 
df.drop_duplicates(subset ="ID", keep = 'first', inplace = True)

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 29450 entries, 4818 to 30228
Data columns (total 5 columns):
ID            29450 non-null int64
Date- Time    29450 non-null object
User_Name     29447 non-null object
Tweet         29450 non-null object
Sentiment     29450 non-null object
dtypes: int64(1), object(4)
memory usage: 1.3+ MB


        after deleting we see the number of tweets / rows in the dataframme re reduced 
        this happens as the duplicate entries are dropped.
        
        after deleting the tweets the index might get changed ,they will contain missing values as rows are deleted.
        we need to reset the index so the the index maintains a serialorder n our dataset.

In [11]:
df.head()

ID           Date- Time         User_Name  \
4818   1286896213657346050  2020-07-25 05:29:01  BALLl              
4817   1286905185273610241  2020-07-25 06:04:40  T:HE:JAMES. HMRH   
2961   1287391022218006531  2020-07-26 14:15:12  Anshi Pathak       
10249  1287495418180501505  2020-07-26 21:10:02  Goodauld69th       
4816   1287731665452769280  2020-07-27 12:48:48  Sardar Khan        

                                                                                                                                                                                                                                                                                                           Tweet  \
4818   i need to make my own SapceX, some people dont deserve to live here                                                                                                                                                                                                                                         
4817   Thre was mention of this earlier, the infraction on the views of the cosmos from the SapceX array and how astronomy would compensate. https://t.co/I5omeFq4SM                                                                                                                                               
2961   @bhab005 @KanganaTeam @kunalpande8 Sachine Tendulkar handle his account and but here kangana team handles her. That's the difference here. And if she is doing that what's funny. She has earned that right. Also Jo log khud nahi bolte h wo dusron se bulwate Hain. Wo khud bol rahi h to galat. Kya h.   
10249  Mainly stoltenberg scandanavian cowardly attacks oil pipe lines then running like weird inbred fish savages to underground bunkers i wont dive into it but that world is over and like nazis there being spitefull and abusive and fucking everything up because there dumb inbreds                         
4816   @gareebscientist @GauravG91 @Chandel1402 @isro @SpaceX He I mean... Vaibhav....spacex is doing great work.. But its unfair to compare two.. No offense to sapcex but I will chose ISRO over spacex AnyDay....                                                                                               

      Sentiment  
4818   positive  
4817   neutral   
2961   positive  
10249  negative  
4816   negative

In [14]:
df=df.reset_index(drop= True)
df.head()

ID           Date- Time         User_Name  \
0  1286896213657346050  2020-07-25 05:29:01  BALLl              
1  1286905185273610241  2020-07-25 06:04:40  T:HE:JAMES. HMRH   
2  1287391022218006531  2020-07-26 14:15:12  Anshi Pathak       
3  1287495418180501505  2020-07-26 21:10:02  Goodauld69th       
4  1287731665452769280  2020-07-27 12:48:48  Sardar Khan        

                                                                                                                                                                                                                                                                                                       Tweet  \
0  i need to make my own SapceX, some people dont deserve to live here                                                                                                                                                                                                                                         
1  Thre was mention of this earlier, the infraction on the views of the cosmos from the SapceX array and how astronomy would compensate. https://t.co/I5omeFq4SM                                                                                                                                               
2  @bhab005 @KanganaTeam @kunalpande8 Sachine Tendulkar handle his account and but here kangana team handles her. That's the difference here. And if she is doing that what's funny. She has earned that right. Also Jo log khud nahi bolte h wo dusron se bulwate Hain. Wo khud bol rahi h to galat. Kya h.   
3  Mainly stoltenberg scandanavian cowardly attacks oil pipe lines then running like weird inbred fish savages to underground bunkers i wont dive into it but that world is over and like nazis there being spitefull and abusive and fucking everything up because there dumb inbreds                         
4  @gareebscientist @GauravG91 @Chandel1402 @isro @SpaceX He I mean... Vaibhav....spacex is doing great work.. But its unfair to compare two.. No offense to sapcex but I will chose ISRO over spacex AnyDay....                                                                                               

  Sentiment  
0  positive  
1  neutral   
2  positive  
3  negative  
4  negative

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29450 entries, 0 to 29449
Data columns (total 5 columns):
ID            29450 non-null int64
Date- Time    29450 non-null object
User_Name     29447 non-null object
Tweet         29450 non-null object
Sentiment     29450 non-null object
dtypes: int64(1), object(4)
memory usage: 1.1+ MB


# Check if any duplicate tweets remians based on Tweet id.

In [16]:
duplicateRowsDF = df[df.duplicated(['ID'])]
duplicateRowsDF

Empty DataFrame
Columns: [ID, Date- Time, User_Name, Tweet, Sentiment]
Index: []

# if no Duplicates append tweets to final csv file

        run with caution: do not run twice for same set of tweets

In [17]:
os.remove("TweetlabelledFinal.csv")

In [18]:
# if file does not exist write header 
if not os.path.isfile('TweetlabelledFinal.csv'):
   df.to_csv('TweetlabelledFinal.csv', header='column_names',index=False)
else: # else it exists so append without mentioning the header
   df.to_csv('TweetlabelledFinal.csv', mode='a', header=False,index=False)

# check if final CSV Has any Duplicates

In [19]:
df= pd.read_csv('TweetlabelledFinal.csv')

In [20]:
duplicateRowsDF = df[df.duplicated(['ID'])]
duplicateRowsDF

Empty DataFrame
Columns: [ID, Date- Time, User_Name, Tweet, Sentiment]
Index: []

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29450 entries, 0 to 29449
Data columns (total 5 columns):
ID            29450 non-null int64
Date- Time    29450 non-null object
User_Name     29447 non-null object
Tweet         29450 non-null object
Sentiment     29450 non-null object
dtypes: int64(1), object(4)
memory usage: 1.1+ MB


In [22]:
df.tail(10)

ID           Date- Time                    User_Name  \
29440  1324044314910011393  2020-11-04 17:42:18  Starts with G 🇪🇺🏳️‍🌈          
29441  1324044319792267268  2020-11-04 17:42:20  3fatcats ازادی                
29442  1324044322589822979  2020-11-04 17:42:20  MJC (PARLER) gelicmonkey      
29443  1324044323659264001  2020-11-04 17:42:20  Leki007 🔱 💫 🦋                 
29444  1324044326272466944  2020-11-04 17:42:21  Carmen Sanchez                
29445  1324044340285485056  2020-11-04 17:42:24  Nupur J Sharma                
29446  1324044349810892802  2020-11-04 17:42:27  servicedogtrainer 🔮🐕‍🦺🌊       
29447  1324044350196604928  2020-11-04 17:42:27  Bhavna Thakur                 
29448  1324044371424026629  2020-11-04 17:42:32  FBI Trader Media              
29449  1324044377749028864  2020-11-04 17:42:33  Nafis Siddiqui (نفیس صدیقی)   

                                                                                                                                                                                                                                                              Tweet  \
29440  Me in October 2016 vs me after four years of Trump administration + Election Night 2020 https://t.co/umBMVksZp7                                                                                                                                                
29441  Georgia still has at least 240,000 votes left to count, secretary of state says (too close to call) https://t.co/LlJK3be9To                                                                                                                                    
29442  URGENT. AYONE  WHO WAS TOLD  USE SHARPIES GET IN TOUCH W @JaySekulow  HE NEEDS TO HEAR FROM YOU. CALL HIM NOW Watch @JaySekulow's broadcast: Radio: ELECTION 2020: Too Close to Call, Legal Battle Begins\n\nhttps://t.co/4fwR8tEIiJ https://t.co/soy8JOgxNB   
29443  11:11 \nPraying for Arnab Goswami to be safe &amp; sound\nPraying for Republic TV staff\nPraying for freedom of Speech \nPraying for Justice\n#IndiaWithArnab 🌟🌟🌟                                                                                              
29444  Don't be fooled: the delays in the US election result mean our system is working | Lawrence Douglas https://t.co/XDx6OAnWIU                                                                                                                                    
29445  BJP leader Ram Kadam to meet the Maharashtra Governor  at 10:30 AM tomorrow to discuss the treatment meted out to Arnab Goswami                                                                                                                                
29446  @Gold_Ranga Gained one here! Black lives matter! PoC lives matter!                                                                                                                                                                                             
29447  @Tejasvi_Youth Baat achhi hai but BJP needs to ensure his safety and well-being. Congress minister has already cleared that one team is formed only to harass Arnab Goswami.                                                                                   
29448  Mohamed Salah rates Liverpool's Champions League chances as Rio Ferdinand backs Reds https://t.co/pbWjXMZTh0                                                                                                                                                   
29449  People used to think Arnab was a spokesman of the BJP.\n\n Well, People were wrong.\n\n BJP is the spokesperson of Arnab Goswami.                                                                                                                              

      Sentiment  
29440  neutral   
29441  negative  
29442  positive  
29443  positive  
29444  negative  
29445  neutral   
29446  negative  
29447  neutral   
29448  neutral   
29449  negative

In [23]:
df.groupby(['Sentiment']).count()

ID  Date- Time  User_Name  Tweet
Sentiment                                     
negative   9485   9485        9485       9485 
neutral    9627   9627        9624       9627 
positive   10338  10338       10338      10338

# Run this part only after you have collected all tweets you need 

    we will check for duplicate tweets based on the tweets itself, till now we were checking based on tweet id
    but some user can repost the same tweet hence run this part to take care of any duplicates

In [24]:
last_check= pd.read_csv('TweetlabelledFinal.csv')

In [25]:
FinalduplicateRows = last_check[last_check.duplicated(['Tweet'])]
FinalduplicateRows

ID           Date- Time  \
24     1288131373010522114  2020-07-28 15:17:06   
33     1288312634421284864  2020-07-29 03:17:22   
44     1288584362267521024  2020-07-29 21:17:07   
56     1288765597501632513  2020-07-30 09:17:17   
68     1289037391081029633  2020-07-31 03:17:17   
171    1289309137311080455  2020-07-31 21:17:07   
312    1289490369881583616  2020-08-01 09:17:16   
322    1289494624835211269  2020-08-01 09:34:10   
476    1289546536972517379  2020-08-01 13:00:27   
1040   1289608877756108801  2020-08-01 17:08:10   
1339   1289622444202811392  2020-08-01 18:02:05   
1983   1289636960869339136  2020-08-01 18:59:46   
2904   1289646455527940096  2020-08-01 19:37:30   
4365   1289651334421966848  2020-08-01 19:56:53   
4467   1289651746969509888  2020-08-01 19:58:31   
4504   1289651884349784065  2020-08-01 19:59:04   
6308   1289660210844299264  2020-08-01 20:32:09   
6507   1289660816971534336  2020-08-01 20:34:34   
7509   1289762170423582722  2020-08-02 03:17:18   
7566   1290033909548240896  2020-08-02 21:17:06   
7627   1290215149593190404  2020-08-03 09:17:17   
7696   1290262086715547648  2020-08-03 12:23:48   
7798   1290286323228479492  2020-08-03 14:00:06   
8113   1290296450472005633  2020-08-03 14:40:20   
8852   1290301328384643073  2020-08-03 14:59:43   
9695   1290302815794016256  2020-08-03 15:05:38   
11233  1290486949543686146  2020-08-04 03:17:19   
11609  1290758685438775296  2020-08-04 21:17:06   
11647  1290769450128617479  2020-08-04 21:59:52   
11956  1290864334239756288  2020-08-05 04:16:55   
...                    ...                  ...   
22856  1291369135650136065  2020-08-06 13:42:49   
23974  1291377897362661376  2020-08-06 14:17:38   
24044  1291377930292072448  2020-08-06 14:17:45   
24303  1291380438766821376  2020-08-06 14:27:43   
24325  1291380583189241856  2020-08-06 14:28:18   
24333  1291380616059957251  2020-08-06 14:28:26   
24381  1291380901843148805  2020-08-06 14:29:34   
24762  1291381445114634240  2020-08-06 14:31:43   
24770  1291381485031895042  2020-08-06 14:31:53   
25081  1291383705982971904  2020-08-06 14:40:42   
25578  1291391057926729728  2020-08-06 15:09:55   
25870  1291391846992805888  2020-08-06 15:13:03   
28588  1324038910834520066  2020-11-04 17:20:50   
28622  1324039620527620096  2020-11-04 17:23:39   
28624  1324039654216224771  2020-11-04 17:23:47   
28629  1324039773988802562  2020-11-04 17:24:16   
28772  1324042702497959938  2020-11-04 17:35:54   
28774  1324042703395528706  2020-11-04 17:35:54   
28775  1324042704406405126  2020-11-04 17:35:54   
28776  1324042705358462976  2020-11-04 17:35:55   
28777  1324042706272849921  2020-11-04 17:35:55   
28778  1324042707153670147  2020-11-04 17:35:55   
28779  1324042708076404745  2020-11-04 17:35:55   
29044  1324043959866400768  2020-11-04 17:40:54   
29045  1324043960780730369  2020-11-04 17:40:54   
29046  1324043961774858242  2020-11-04 17:40:54   
29047  1324043962693439488  2020-11-04 17:40:54   
29048  1324043963624546310  2020-11-04 17:40:55   
29049  1324043964635320326  2020-11-04 17:40:55   
29050  1324043965553922049  2020-11-04 17:40:55   

                                 User_Name  \
24     Albus Dumbledore                      
33     Albus Dumbledore                      
44     Albus Dumbledore                      
56     Albus Dumbledore                      
68     Albus Dumbledore                      
171    Albus Dumbledore                      
312    Albus Dumbledore                      
322    Sabras Radio                          
476    Anuj Chase                            
1040   Sabras Radio                          
1339   Toronto Fire West                     
1983   J-Script                              
2904   Rakesh kumar Singh राकेश कुमार सिंह   
4365   Lisle‎†                               
4467   ً                                     
4504   ً                                     
6308   Toronto Fire South                    
6507   Statewide CTDOT           

In [26]:
last_check.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29450 entries, 0 to 29449
Data columns (total 5 columns):
ID            29450 non-null int64
Date- Time    29450 non-null object
User_Name     29447 non-null object
Tweet         29450 non-null object
Sentiment     29450 non-null object
dtypes: int64(1), object(4)
memory usage: 1.1+ MB


In [27]:
last_check.drop_duplicates(subset ="Tweet", keep = 'first', inplace = True)

In [28]:
last_check.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 29348 entries, 0 to 29449
Data columns (total 5 columns):
ID            29348 non-null int64
Date- Time    29348 non-null object
User_Name     29345 non-null object
Tweet         29348 non-null object
Sentiment     29348 non-null object
dtypes: int64(1), object(4)
memory usage: 1.3+ MB


In [29]:
FinalduplicateRows = last_check[last_check.duplicated(['Tweet'])]
FinalduplicateRows

Empty DataFrame
Columns: [ID, Date- Time, User_Name, Tweet, Sentiment]
Index: []

In [30]:
FinalduplicateRows = last_check[last_check.duplicated(['ID'])]
FinalduplicateRows

Empty DataFrame
Columns: [ID, Date- Time, User_Name, Tweet, Sentiment]
Index: []

# By now surely all duplicate rows have been removed 

    so save the final Dataframe "last_check" in your csv file

In [31]:
os.remove("TweetlabelledFinal.csv")
#if file does not exist write header 
if not os.path.isfile('TweetlabelledFinal.csv'):
   last_check.to_csv('TweetlabelledFinal.csv', header='column_names',index=False)
else: # else it exists so append without mentioning the header
   last_check.to_csv('TweetlabelledFinal.csv', mode='a', header=False,index=False)

# Read your Final tweet Collection Csv file and check it 

In [32]:
tweets = pd.read_csv('TweetlabelledFinal.csv')

In [33]:
pd.set_option('display.max_colwidth', -1) # Setting this so we can see the full content of cells

In [34]:
tweets.head()

ID           Date- Time         User_Name  \
0  1286896213657346050  2020-07-25 05:29:01  BALLl              
1  1286905185273610241  2020-07-25 06:04:40  T:HE:JAMES. HMRH   
2  1287391022218006531  2020-07-26 14:15:12  Anshi Pathak       
3  1287495418180501505  2020-07-26 21:10:02  Goodauld69th       
4  1287731665452769280  2020-07-27 12:48:48  Sardar Khan        

                                                                                                                                                                                                                                                                                                       Tweet  \
0  i need to make my own SapceX, some people dont deserve to live here                                                                                                                                                                                                                                         
1  Thre was mention of this earlier, the infraction on the views of the cosmos from the SapceX array and how astronomy would compensate. https://t.co/I5omeFq4SM                                                                                                                                               
2  @bhab005 @KanganaTeam @kunalpande8 Sachine Tendulkar handle his account and but here kangana team handles her. That's the difference here. And if she is doing that what's funny. She has earned that right. Also Jo log khud nahi bolte h wo dusron se bulwate Hain. Wo khud bol rahi h to galat. Kya h.   
3  Mainly stoltenberg scandanavian cowardly attacks oil pipe lines then running like weird inbred fish savages to underground bunkers i wont dive into it but that world is over and like nazis there being spitefull and abusive and fucking everything up because there dumb inbreds                         
4  @gareebscientist @GauravG91 @Chandel1402 @isro @SpaceX He I mean... Vaibhav....spacex is doing great work.. But its unfair to compare two.. No offense to sapcex but I will chose ISRO over spacex AnyDay....                                                                                               

  Sentiment  
0  positive  
1  neutral   
2  positive  
3  negative  
4  negative

In [35]:
tweets.tail()

ID           Date- Time                    User_Name  \
29343  1324044340285485056  2020-11-04 17:42:24  Nupur J Sharma                
29344  1324044349810892802  2020-11-04 17:42:27  servicedogtrainer 🔮🐕‍🦺🌊       
29345  1324044350196604928  2020-11-04 17:42:27  Bhavna Thakur                 
29346  1324044371424026629  2020-11-04 17:42:32  FBI Trader Media              
29347  1324044377749028864  2020-11-04 17:42:33  Nafis Siddiqui (نفیس صدیقی)   

                                                                                                                                                                              Tweet  \
29343  BJP leader Ram Kadam to meet the Maharashtra Governor  at 10:30 AM tomorrow to discuss the treatment meted out to Arnab Goswami                                                
29344  @Gold_Ranga Gained one here! Black lives matter! PoC lives matter!                                                                                                             
29345  @Tejasvi_Youth Baat achhi hai but BJP needs to ensure his safety and well-being. Congress minister has already cleared that one team is formed only to harass Arnab Goswami.   
29346  Mohamed Salah rates Liverpool's Champions League chances as Rio Ferdinand backs Reds https://t.co/pbWjXMZTh0                                                                   
29347  People used to think Arnab was a spokesman of the BJP.\n\n Well, People were wrong.\n\n BJP is the spokesperson of Arnab Goswami.                                              

      Sentiment  
29343  neutral   
29344  negative  
29345  neutral   
29346  neutral   
29347  negative

In [36]:
DuplicateRows = tweets[tweets.duplicated(['ID'])]
DuplicateRows

Empty DataFrame
Columns: [ID, Date- Time, User_Name, Tweet, Sentiment]
Index: []

In [37]:
DuplicateRows = tweets[tweets.duplicated(['Tweet'])]
DuplicateRows

Empty DataFrame
Columns: [ID, Date- Time, User_Name, Tweet, Sentiment]
Index: []

In [38]:
tweets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29348 entries, 0 to 29347
Data columns (total 5 columns):
ID            29348 non-null int64
Date- Time    29348 non-null object
User_Name     29345 non-null object
Tweet         29348 non-null object
Sentiment     29348 non-null object
dtypes: int64(1), object(4)
memory usage: 1.1+ MB


In [39]:
tweets.groupby(['Sentiment']).count()

ID  Date- Time  User_Name  Tweet
Sentiment                                     
negative   9468   9468        9468       9468 
neutral    9587   9587        9584       9587 
positive   10293  10293       10293      10293

# The final Csv File of tweets has no duplicate entries, and is ready for use

    the final file in this case "TweetslabelledFinal" conatins:
    Total Tweets : 29,348
    fields : Tweet ID , Date-Time, User Name, Tweet text and Sentiment
            Positive Tweets : 10,293
            Negative Tweets : 9,468
            Neutral Tweets :  9,587